app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -58,22 +58,17 @@
 
     private final static String TAG = "SmsDetector";
 
-    private DataInputStream dis;
     private DataOutputStream dos;
+    BufferedReader in;
     private AimsicdService mAimsicdService;
-    private SharedPreferences prefs;
     private boolean mBound;
     private AIMSICDDbAdapter dbacess;
     private Context tContext;
-    private String[] SILENT_ONLY_TAGS;
+    private String[] LOADED_DETECTION_STRINGS;
+    private int B_LINED_SIZE = 30;// how many previous lines to hold
+    private String[] BUFFEREDLINES = new String[B_LINED_SIZE];//for holding previous lines from logcat that hold data we need
+    private final int TYPE0 = 1,MWI =2,WAP =3;
```

Comment:
it's a few hundred bytes here, but think if you have a class that has more class attributes inside, then memory size of one field which COULD BE static grows really fast. Or if you have a bitmap, that's a really memory consumings class, there is a bug in android < 5 where bitmap.c is leaking memory, it's easy to fix that bug with your code by defining it static. The bug is in .c file so Dalvik System Garbage Collector won't free memory after it leaked.